exposed-core/src/main/kotlin/org/jetbrains/exposed/dao/id/IdTable.kt

```diff
@@ -34,6 +34,71 @@ object EntityIDFunctionProvider {
 abstract class IdTable<T : Comparable<T>>(name: String = "") : Table(name) {
     /** The identity column of this [IdTable], for storing values of type [T] wrapped as [EntityID] instances. */
     abstract val id: Column<EntityID<T>>
+
+    /** All base columns that make up this [IdTable]'s identifier column. */
+    val idColumns = HashSet<Column<out Comparable<*>>>()
+
+    /**
+     * Returns a boolean operator comparing each of this table's [idColumns] to its corresponding
+     * value in [toCompare], using the specified SQL [operator].
+     *
+     * @throws IllegalStateException If [toCompare] does not contain a key for each component column.
+     */
+    @Suppress("UNCHECKED_CAST")
+    internal fun mapIdComparison(
+        toCompare: Any?,
+        operator: (Column<*>, Expression<*>) -> Op<Boolean>
+    ): Op<Boolean> {
+        return if (idColumns.size == 1 && this !is CompositeIdTable) {
```

Comment:
It looks like there is mostly a choice between a single id table and composite id table. 

`mapIdComparison()` could be overridden inside the `CompositeIdTable` with the specific for that class logic.

In this case inside `IdTable` would be simpler method:

```kotlin
internal open fun mapIdComparison(
    toCompare: Any?,
    operator: (Column<*>, Expression<*>) -> Op<Boolean>
): Op<Boolean> {
    val singleId = idColumns.single()
    return operator(singleId, singleId.wrap(toCompare))
}
```

and inside `CompositeIdTable`:

```kotlin
override fun mapIdComparison(toCompare: Any?, operator: (Column<*>, Expression<*>) -> Op<Boolean>): Op<Boolean> {
    (toCompare as? EntityID<CompositeID>) ?: error("toCompare must be EntityID<CompositeID>")
    return idColumns.map { column ->
        val otherValue = if (column in toCompare.value.values) {
            toCompare.value[column as Column<EntityID<Comparable<Any>>>]
        } else {
            error("Comparison CompositeID is missing a key mapping for ${column.name}")
        }
        operator(column, column.wrap(otherValue))
    }.compoundAnd()
}
```

Reply:
Leaving this temporarily until we confirm how to validate that `IdTable` only ever has 1 `idColumn`.